### DISCUSSION TITLE
How to add SCTE-35 Markers on Decklink > ffmpeg  > UDP stream ?

### DISCUSSION THREAD

**Discussion by gaamaaresosa** [futzu/threefive#125] [Q&A] [answered]
Hi Adrian,

I read everyone suggest threefive for SCTE-35.

I am very new to SCTE-35 concepts.
Please help me to get into it.
(I use Windows 10 64bit OS)
(VisualBasic 6 is my coding language)

My Client application play videos and render on Decklink Card successfully.
On the same system, my encoder application get that video stream from Decklink capture card and use ffmpeg to stream out on udp:// or srt:// to the remote server.

The remote (My Server) application receives that stream and playing there successfully.
So far everything works fine.

Now how to inject SCTE-35 markers on the above chain and send UDP:// or SRT:// with embedded SCTE-35 markers ?
So that the server will detect it and switch local ads there.

I wish some Text/XML file as a middleware, which I need to update (Overwrite) from my application.
ffmpeg/threefive should dynamically read that Text/XML file and inject the SCTE-35 markers on the stream and send.

Please give me the workflow with your tools.
Also please give me sample command line example.

No where I could find sample for Window with ffmpeg.

Thanks in advance !
gaamaa

**Comment by futzu** [comment]
That is something you can hire me to do for you.

Adrian

**Comment by gaamaaresosa** [comment]
Approximately how much will you charge ?

⁣gaamaa​

**Comment by futzu** [comment]
For something like that,  $100 an hour.
I don't have time talk right now, we can discuss it further on Monday if you like.

**Comment by futzu** [comment]
I guess you don't need me so I'm going to close this. Just open an issue if you need me.

Adrian

**Comment by gaamaaresosa** [comment]
I am really sorry to say that I don't have budget for this project.
I need to find some open source solutions.

Thanks again 

⁣gaamaa​

**Reply by futzu** [reply, marked as answer]
I think there's more involved than you realize. In concept it's fairly simple, 
but implementation tends to more difficult than expected.
 I could it in 10 -15 hours, assuming there aren't any issues.
you've probably have 15 hours on it since Saturday.

**Comment by gaamaaresosa** [comment]
Thank you so much again.

First of all even I don't know what I wanted.

I play video files and render it in Decklink card. Also on virtual directshow filter.

I did it in in C++ 3rd party ocx and my own VB6 coding on Windows 10 64bit system.

Also from VB6 I use ffmpeg command line and pass above Virtual filter/Deklink as source, streaming to remote server as srt:// or rtmp://

This works great since many years.

Now my client needs SCET-35 AD markers injection when I play ads on my playout software. So they will switch their regional ads there.

I don't know how to do in on my existing rtmp:// or srt:// stream.

Or any better logical streaming to be used.

Still learning the concepts.

⁣gaamaa​
